app/models/workflow.rb

```diff
@@ -5,6 +5,7 @@ class Workflow < ApplicationRecord
   has_many :requests, -> { order(:id => :asc) }, :inverse_of => :workflow
 
   validates :name, :presence => :name
+  validates :name, :uniqueness => { :scope => :tenant_id }
```

Comment:
should combine two lines into one:
```
validates :name, :presence => true, :uniqueness => { :scope => :tenant_id }
```
However this scope does not prevent a workflow in a tenant from having the same name as another one without a tenant.

Reply:
This was my question yesterday.  Is the `scope` needed at all?  I think we should be comparing to the workflows that are returned which should already be scoped to the tenant + global.

Put a different way, you are only creating workflows within a tenant, with the exception of the globally seeded workflows which are done by the system.

cc @syncrou

Reply:
@gmcculloug I tested. `ActsAsTenant` does not automatically add the scope.

Reply:
Correct, the controller would set the tenant.  My point was that you really are never creating workflows outside of a tenant which would have applied the scope.  (Again, with the exception of global system workflows.)

So if you are testing you need to use the `ActsAsTenant.with_tenant` as part of the test.  In the catalog repo we updated our tests to account for this.

Reply:
We ship a workflow `Always approve` which belongs to no tenant. We need to prevent an admin from creating another `Always approve` workflow. 
We may create a [custom validation method](https://guides.rubyonrails.org/active_record_validations.html#custom-methods)

Reply:
Agreed.  After testing locally I can verify that the validation does not honor the current tenant setting and requires the validation to include the proper tenant scope.  👍